### PR TITLE
Public API changes recommended by the design team for the new Analyze…

### LINKED ISSUE #6044: API Review Followup - Analyzers/Suppression
These are the remaining items from our 1.1 API review:

- CompilationWithAnalyzerOptions should be sealed
- CompilationWithAnalyzerOptions.AnalyzerOptions should be renamed Options
- SuppressionInfo should be sealed
- AnalyzerTelemetry API changes as per [this](https://github.com/dotnet/roslyn/issues/5974#issuecomment-149987997) comment.

### PR DESCRIPTION
**Pull request by mavasani**
…r/Supression APIs.

- CompilationWithAnalyzerOptions should be sealed
- CompilationWithAnalyzerOptions.AnalyzerOptions should be renamed Options
- SuppressionInfo should be sealed
- AnalyzerTelemetry API changes as per [this](https://github.com/dotnet/roslyn/issues/5974#issuecomment-149987997) comment.

There are no functionality changes here.

Fixes #6044

### CODE REVIEW

File: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -25,8 +25,9 @@ Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.GetAnalyzerSemanticD
...
 Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.GetAnalyzerSyntaxDiagnosticsAsync(Microsoft.CodeAnalysis.SyntaxTree tree, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer> analyzers, System.Threading.CancellationToken cancellationToken) -> System.Threading.Tasks.Task<System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Diagnostic>>
 Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.GetAnalyzerSyntaxDiagnosticsAsync(Microsoft.CodeAnalysis.SyntaxTree tree, System.Threading.CancellationToken cancellationToken) -> System.Threading.Tasks.Task<System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Diagnostic>>
+Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.GetAnalyzerTelemetryInfoAsync(Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer analyzer, System.Threading.CancellationToken cancellationToken) -> System.Threading.Tasks.Task<Microsoft.CodeAnalysis.Diagnostics.Telemetry.AnalyzerTelemetryInfo>
```
Comment (mavasani) [inline src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: This is the primary file to review.

**Comment by mavasani** [comment]
@DustinCampbell @heejaechang @mattwar @srivatsn can you please review?

**Comment by srivatsn** [comment]
:+1:

**Comment by mavasani** [comment]
@dotnet-bot retest  prtest/win/dbg/unit32 please

**Comment by mavasani** [comment]
@dotnet-bot retest prtest/win/dbg/unit64 please

**Comment by mavasani** [comment]
@dotnet-bot retest win unit please.

**Comment by mavasani** [comment]
@dotnet-bot retest win unit please.

**Comment by heejaechang** [comment]
:+1:

**Comment by MattGertz** [comment]
Approved pending successful testing.

**Pull request merged**